editor/src/clj/editor/gui.clj

```diff
@@ -661,11 +746,190 @@
         :type type ; Explicitly include the type (pb-field is optional, so :type-box would be stripped otherwise).
         :child-index child-index))) ; Used to order sibling nodes in the SceneDesc.
 
+;; SDK api
+(defmacro layout-property-getter [prop-sym]
+  {:pre [(symbol? prop-sym)]}
+  (let [prop-kw (keyword prop-sym)]
+    `(g/fnk [~'prop->value ~prop-sym]
+       (get ~'prop->value ~prop-kw ~prop-sym))))
+
+(defn layout-property-set-fn [_evaluation-context node-id _old-value _new-value]
+  (concat
+    (g/invalidate-output node-id :prop->value)
+    (g/invalidate-output node-id :layout->prop->value)))
```

Comment:
We explicitly invalidate these outputs whenever a layout property is set. They both evaluate properties in a bit of a naughty way using `g/node-value` and `_this` in their `fnks`.

* `prop->value` is used in the visual presentation, and is lighter to produce since it focuses on the `current-layout` only.
* `layout->prop->value` is used to produce `build-targets` and `save-value` only.